Java/core/src/main/java/com/amazon/randomcutforest/tree/IBoundingBox.java

```diff
@@ -29,13 +29,25 @@
 
     boolean contains(P point);
 
-    IBox<P> getMergedBox(P point);
+    // duplicates
+    IBoundingBox<P> copy();
 
-    IBox<P> getMergedBox(IBox<P> otherBox);
+    // below keeps the older box unchanged
+    IBoundingBox<P> getMergedBox(P point);
 
-    BoundingBox convertBoxToDouble();
+    // merges in place
+    IBoundingBox<P> addPoint(P point);
 
-    BoundingBoxFloat convertBoxToFloat();
+    // merges and keeops the older box unchaged
+    IBoundingBox<P> getMergedBox(IBoundingBox<P> otherBox);
+
+    // merges in place
+    IBoundingBox<P> addBox(IBoundingBox<P> otherBox);
+
+    BoundingBox copyBoxToDouble();
+
+    float getMinValueFloat(int i);
```

Comment:
Where are these methods used? Can't we just cast to float there instead?

Reply:
Repeatedly casting values can be an issue. But anyway these are removed.